src/main/java/ch/uzh/ifi/hase/soprafs22/service/GameService.java

```diff
@@ -56,9 +58,15 @@ private Game getGameById(Long id) throws GameNotFoundException {
     /**
      * Returns the list of units whose health got affected by the attack.
      */
-    public List<Unit> unitAttack(Long id, String token, Position attacker, Position defender) {
+    public List<Unit> unitAttack(Long id, String token, AttackCommand attackCommand) {
         try {
-            return getGameById(id).unitAttack(token, attacker, defender);
+            Game game = getGameById(id);
+            Position attacker = attackCommand.getAttacker();
+            Position defender = attackCommand.getDefender();
+            List<Unit> units = game.unitAttack(token, attacker, defender);
+            Position attackerDestination = attackCommand.getAttackerDestination();
+            game.unitMove(token, attacker, attackerDestination);
```

Comment:
I would do first move then attack, as the unit attacks from the new position. This will make it way easier to do in range checking.

Reply:
Is there already a issue for, doing the range checking and if a path exists? Else we can create a new one...

Reply:
My thought behind that was the following:
If I have a unit and want to attack a unit that is out of my attackRange with the current setup it will abort the attack and throw an error.
If I do the move first, then the attack, it will move (its attackRange will change) and by that I will be able to attack a unit out of my attackRange.
Probably poor design on the backend side, but that's how it would go at the moment in my opinion.

Reply:
Do you calculate the attackrange with the movement included?

Reply:
nope.

Reply:
Yes, but in the client you can also move the unit from one point to another, and then attack the units that are in range from the new tile

Reply:
> I would do first move then attack, as the unit attacks from the new position. This will make it way easier to do in range checking.

But you're right. I didn't think about this. especially the range checking for the counterattack.